cranelift/codegen/src/opts/algebraic.isle

```diff
@@ -204,6 +200,17 @@
                       (iconst _ k)))
       (band ty x (iconst ty (imm64_ushr ty (imm64 (ty_mask ty)) k))))
 
+;; For signed shifts, `(x << k) >> k` does sign-extension from `n` bits to
+;; `n+k` bits. In the special case where `x` is the result of either
+;; `sextend` or `uextend` from `n` bits to `n+k` bits, we can implement this
+;; using `sextend`.
+(rule (simplify (sshr wide (ishl _ (uextend _ x @ (value_type narrow)) (iconst _ shift)) (iconst _ shift)))
```

Comment:
This would be a little easier to parse if it were split across multiple lines, IMHO (just a preference!).